### PR TITLE
Fix Issue #125 - Spell of Opening does not unlock doors

### PR DESCRIPTION
**Pull request by crkellen**
This is a fix for #125.
The issue was there was no change to the actual lock status of the door being hit, the Spell of Opening would simply do as it's name implies and "open" the door. This however, puts the door in a intangible state, as it is no longer a physical Entity.

Additionally, I opted to remove the effect of opening the door altogether, as I believe it would be better to leave that up to the Player's discretion. By letting them decide whether or not to open the door, prevents issues where a mob runs in when the Player is not ready (swapping items/spells, or otherwise).

### CODE REVIEW

**Comment by SheridanR** [comment]
Good fix for the bug... the spell should open the door though. if you don't want to let monsters in, don't cast the spell.

**Comment by crkellen** [comment]
I actually reverted that locally. I realized soon after that it opens metal gates. So I fixed it and made it open still. Guessing never pushed the changes. Will do when I get a chance.
null

**Comment by crkellen** [comment]
I went ahead and remedied this issue, originally, I didn't realize the Spell of Opening could actually open Iron gates and the like. I simply assumed it was just a counter spell to Locking, and thus "Opening" was really more just "Unlocking".

It now unlocks, and opens.

**Comment by crkellen** [comment]
I went ahead and extended the formatting update to the rest of the Spell of Opening as well. Therefore IF statements have the spaces between braces and such as it should be.

File: src/magic/actmagic.cpp [hunk 1]
```diff
@@ -1909,6 +1909,8 @@ void actMagicMissile(Entity* my)   //TODO: Verify this function.
 					{
 						if (hit.entity->behavior == &actDoor)
 						{
+                            playSoundEntity(hit.entity, 91, 64);
+                            hit.entity->skill[5] = 0; // Unlock the door.
```
Comment (addictgamer) [inline src/magic/actmagic.cpp]: Out of curiosity, is there any `Sint32& ...` defined for this in Entity.hpp? Like `Sint32& doorLocked`?
The chest has a bunch of these sorts of defines, like `Sint32& chestOpener` IIRC

Comment (crkellen) [reply src/magic/actmagic.cpp]: As far as I can see, there is not. It is however possible that `chest_opener(skill[5])` could be used, however for now, I'll keep it unmodified.

**Comment by addictgamer** [comment]
Merged in #208

**Pull request closed without merging**